src/rest_framework_api_key/models.py

```diff
@@ -40,17 +40,24 @@ def create_key(self, **kwargs: typing.Any) -> typing.Tuple["AbstractAPIKey", str
     def get_usable_keys(self) -> models.QuerySet:
         return self.filter(revoked=False)
 
-    def is_valid(self, key: str) -> bool:
+    def get_from_key(self, key: str) -> "AbstractAPIKey":
         prefix, _, _ = key.partition(".")
-
         queryset = self.get_usable_keys()
 
         try:
             api_key = queryset.get(prefix=prefix)
         except self.model.DoesNotExist:
-            return False
+            raise  # For the sake of being explicit.
 
         if not api_key.is_valid(key):
+            raise self.model.DoesNotExist("Key is not valid.")
+        else:
+            return api_key
+
+    def is_valid(self, key: str) -> bool:
+        try:
+            api_key = self.get_from_key(key)
+        except (self.model.DoesNotExist, ValidationError):
```

Comment:
Can `ValidationError` still be raised here?

Reply:
It can't, great catch! I removed the `ValidationError` handling in 6e88a51d9b403895edad159e1f48df6929576093.